protocol/v2/ssv/validator/committee.go

```diff
@@ -177,6 +196,13 @@ func (c *Committee) StartDuty(logger *zap.Logger, duty *spectypes.CommitteeDuty)
 
 	}
 
+	// Stop all expired committee runners, when new runner is created
+	go func() {
+		if err := c.stopExpiredRunners(logger, duty.Slot); err != nil {
+			logger.Error("couldn't stop expired committee runners", zap.Uint64("current_slot", uint64(duty.Slot)), zap.Error(err))
+		}
+	}()
+
```

Comment:
repeating my previous comment which disappeared: we don't need a goroutine here, we can just remove the lock from stopExpiredRunners and execute it directly here

if we're worried about other callers, then it can be renamed to unsafeStopExpiredRunners() with a comment that a lock is required